x-pack/plugins/lens/public/datasources/text_based/fieldlist_cache.ts

```diff
@@ -4,16 +4,31 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+import { type AggregateQuery, getAggregateQueryMode } from '@kbn/es-query';
 import type { DatatableColumn } from '@kbn/expressions-plugin/public';
+import { getAllColumns } from './utils';
+import type { TextBasedLayerColumn } from './types';
 
 const cachedColumns = new Map<string, DatatableColumn[]>();
 
-export const addColumnsToCache = (query: string, list: DatatableColumn[]) => {
-  const trimmedQuery = query.replaceAll('\n', '').trim();
+export const addColumnsToCache = (query: AggregateQuery, list: DatatableColumn[]) => {
+  const language = getAggregateQueryMode(query);
+  const queryString: string = query[language];
+  const trimmedQuery = queryString.replaceAll('\n', '').trim();
```

Comment:
nit: could pull this operation into a `getKey` function used in both places.

Reply:
Nice idea! Done https://github.com/elastic/kibana/pull/174500/commits/05480bdcbc7e473c07a0d75def7be7aee3f5cec0